api/storage_service.cc

```diff
@@ -588,8 +588,24 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
         return ctx.db.local().commitlog()->active_config().commit_log_location;
     });
 
-    ss::get_token_endpoint.set(r, [&ss] (std::unique_ptr<http::request> req) {
-        return make_ready_future<json::json_return_type>(stream_range_as_array(ss.local().get_token_to_endpoint_map(), [](const auto& i) {
+    ss::get_token_endpoint.set(r, [&ctx, &ss] (std::unique_ptr<http::request> req) {
+        const auto keyspace_name = req->get_query_param("keyspace");
+        const auto table_name = req->get_query_param("cf");
+
+        std::map<dht::token, gms::inet_address> token_endpoints;
+        if (keyspace_name.empty() && table_name.empty()) {
+            token_endpoints = ss.local().get_token_to_endpoint_map();
+        } else if (!keyspace_name.empty() && !table_name.empty()) {
+            auto& db = ctx.db.local();
+            if (!db.has_schema(keyspace_name, table_name)) {
+                throw bad_param_exception(fmt::format("Failed to find table {}.{}", keyspace_name, table_name));
+            }
```

Comment:
should we document that cf param is restricted to tablets? or perhaps fall back to vnode mode if we detect table is not tablet based?

Reply:
Currently, with keyspace and table set, the API endpoint will fail with failure to find tablet map for table.
I don't think we should make this silently fall-back to vnodes.

Reply:
I expanded the docs a bit